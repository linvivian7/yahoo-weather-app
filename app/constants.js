export const LOCATION_FORM_NAME = 'locationForm';
