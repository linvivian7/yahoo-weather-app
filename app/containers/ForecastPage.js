import React from 'react';

import Nav from '../components/nav';
import ForecastCard from '../components/forecastCard';

export default class ForecastPage extends React.Component {

    shouldComponentUpdate(nextProps) {
        return nextProps.weatherInfo !== this.props.weatherInfo;
    }

    onHomePageLinkClick(e) {
        e.preventDefault();

        this.props.onHomePageLinkClick();
    }

    onForecastLinkClick(e) {
        e.preventDefault();

        this.props.onForecastLinkClick();
    }

    _getListItems(items) {
        return items.map((item, index) => (
            <ForecastCard key={ `forecast${item.date}` } item={ item } index={ index } />
        ));
    }

    _getForecastContainer(forecast) {
        console.log(forecast);
        return (
            <ul className="forecast-page-container">
                { this._getListItems(forecast) }
            </ul>
        );
    }

    render() {
        const {
            onChange,
            onSubmit,
            searchTerm,
            weatherInfo
        } = this.props;

        return (
            <div>
                <Nav
                    location={ searchTerm }
                    onChange={ onChange }
                    onSubmit={ onSubmit }
                    onHomePageLinkClick={ this.onHomePageLinkClick.bind(this) }
                    onForecastLinkClick={ this.onForecastLinkClick.bind(this) }
                />
                { this._getForecastContainer(weatherInfo.item.forecast) }
            </div>
        );
    }
}
