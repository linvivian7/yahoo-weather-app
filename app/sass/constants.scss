@import url('https://fonts.googleapis.com/css?family=Roboto+Condensed');

$break-width: 768px;
$temp-hover-color: #2196F3;

$text-color-sandstone: #F4CC70;
$text-color-burnt-orange: #DE7A22;
$text-color-sea: #20948B;
$text-color-lagoon: #6AB187;

$font-family-primary: 'Roboto Condensed', sans-serif;
$font-family-nav: 'Open Sans', sans-serif;

$font-size-small: 16px;
$font-size-medium: 24px;
$font-size-med-large: 60px;
$font-size-large: 96px;
$font-size-x-large: 112px;

$icon-size-x-small: 32px;
$icon-size-small: 48px;
$icon-size-medium: 60px;
$icon-size-x-large: 120px;

$padding-x-small: 4px;
$padding-small: 16px;
$padding-medium: 24px;
$padding-med-large: 36px;
$padding-large: 48px;

$margin-x-small: 4px;
$margin-small: 16px;
$margin-medium: 24px;
$margin-med-large: 36px;
$margin-large: 48px;

@mixin status-card {
    box-shadow: 0 2.5px 8px 0 rgba(0,0,0,.2);
    color: white;
    display: grid;
    font-family: $font-family-primary;
    margin-bottom: 15px 0;
    align-items: center;
}
