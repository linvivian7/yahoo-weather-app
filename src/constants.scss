@import url('https://fonts.googleapis.com/css?family=Roboto+Condensed');

$break-width: 768px;

$color-grey: #AAA;
$color-sunrise: #EAECC6;
$color-sandstone: #F4CC70;
$color-burnt-orange: #DE7A22;
$color-lagoon: #6AB187;
$color-sea: #20948B;

$font-family-primary: 'Roboto Condensed', sans-serif;
$font-family-nav: 'Open Sans', sans-serif;

$font-size-x-small: 12px;
$font-size-small: 3vw;
$font-size-medium: 4vw;
$font-size-med-large: 64px;
$font-size-large: 96px;
$font-size-x-large: 112px;

$icon-size-x-small: 32px;
$icon-size-small: 48px;
$icon-size-medium: 60px;
$icon-size-x-large: 120px;

$padding-x-small: 4px;
$padding-small: 16px;
$padding-medium: 24px;
$padding-med-large: 36px;
$padding-large: 48px;

$margin-x-small: 4px;
$margin-small: 16px;
$margin-medium: 24px;
$margin-med-large: 36px;
$margin-large: 48px;

@mixin status-card {
    box-shadow: 0 2.5px 8px 0 rgba(0,0,0,.2);
    color: white;
    display: grid;
    font-family: $font-family-primary;
    margin-bottom: 15px 0;
    align-items: center;

    @media (min-width: $break-width) {
        padding: 1em;
    }
}
