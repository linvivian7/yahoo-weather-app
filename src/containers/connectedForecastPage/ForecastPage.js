import React from 'react';

import Menu from '../../components/menu';
import ForecastCard from '../../components/forecastCard';
import { getIsMetric } from '../../utils/measurement';

import './forecastPage.scss';

export default class ForecastPage extends React.PureComponent {

    _getListItems(items, isMetric) {
        return items.map((item, index) => (
            <ForecastCard key={ `forecast${item.date}` } item={ item } index={ index } isMetric={ isMetric } />
        ));
    }

    _getForecastContainer(forecast, isMetric) {
        return (
            <ul className="forecast-page-container">
                { this._getListItems(forecast, isMetric) }
            </ul>
        );
    }

    render() {
        const {
            onChange,
            onSubmit,
            onHomePageLinkClick,
            onForecastLinkClick,
            searchTerm,
            weatherInfo
        } = this.props;
        const isMetric = getIsMetric(weatherInfo.units);

        return (
            <div>
                <Menu
                    searchTerm={ searchTerm }
                    onChange= { onChange }
                    onSubmit= { onSubmit }
                    onHomePageLinkClick={ onHomePageLinkClick }
                    onForecastLinkClick={ onForecastLinkClick }
                />
                { this._getForecastContainer(weatherInfo.item.forecast, isMetric) }
            </div>
        );
    }
}
