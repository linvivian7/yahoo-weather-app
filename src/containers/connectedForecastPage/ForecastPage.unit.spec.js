import React from 'react';
import ForecastPage from './ForecastPage';

import * as constants from '../../test/constants';

test('ForecastPage snapshot', () => {
    const wrapper = shallow(
        <ForecastPage
            searchTerm={ constants.SEARCH_TERM }
            weatherInfo={ constants.WEATHER_INFO }
        />
    );

    expect(wrapper).toMatchSnapshot();
});
