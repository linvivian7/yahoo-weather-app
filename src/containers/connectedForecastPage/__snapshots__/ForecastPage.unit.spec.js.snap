// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastPage snapshot 1`] = `
<div>
  <Menu
    searchTerm="Taipei, Taiwan"
  />
  <ul
    className="forecast-page-container"
  >
    <ForecastCard
      index={0}
      isMetric={false}
      item={
        Object {
          "code": "12",
          "date": "21 Apr 2018",
          "day": "Sat",
          "high": "86",
          "low": "71",
        }
      }
      key="forecast21 Apr 2018"
    />
    <ForecastCard
      index={1}
      isMetric={false}
      item={
        Object {
          "code": "4",
          "date": "22 Apr 2018",
          "day": "Sun",
          "high": "88",
          "low": "73",
        }
      }
      key="forecast22 Apr 2018"
    />
    <ForecastCard
      index={2}
      isMetric={false}
      item={
        Object {
          "code": "4",
          "date": "23 Apr 2018",
          "day": "Mon",
          "high": "88",
          "low": "73",
        }
      }
      key="forecast23 Apr 2018"
    />
    <ForecastCard
      index={3}
      isMetric={false}
      item={
        Object {
          "code": "47",
          "date": "24 Apr 2018",
          "day": "Tue",
          "high": "81",
          "low": "72",
        }
      }
      key="forecast24 Apr 2018"
    />
    <ForecastCard
      index={4}
      isMetric={false}
      item={
        Object {
          "code": "28",
          "date": "25 Apr 2018",
          "day": "Wed",
          "high": "74",
          "low": "68",
        }
      }
      key="forecast25 Apr 2018"
    />
    <ForecastCard
      index={5}
      isMetric={false}
      item={
        Object {
          "code": "30",
          "date": "26 Apr 2018",
          "day": "Thu",
          "high": "79",
          "low": "67",
        }
      }
      key="forecast26 Apr 2018"
    />
    <ForecastCard
      index={6}
      isMetric={false}
      item={
        Object {
          "code": "30",
          "date": "27 Apr 2018",
          "day": "Fri",
          "high": "82",
          "low": "69",
        }
      }
      key="forecast27 Apr 2018"
    />
    <ForecastCard
      index={7}
      isMetric={false}
      item={
        Object {
          "code": "34",
          "date": "28 Apr 2018",
          "day": "Sat",
          "high": "83",
          "low": "71",
        }
      }
      key="forecast28 Apr 2018"
    />
    <ForecastCard
      index={8}
      isMetric={false}
      item={
        Object {
          "code": "30",
          "date": "29 Apr 2018",
          "day": "Sun",
          "high": "84",
          "low": "72",
        }
      }
      key="forecast29 Apr 2018"
    />
    <ForecastCard
      index={9}
      isMetric={false}
      item={
        Object {
          "code": "30",
          "date": "30 Apr 2018",
          "day": "Mon",
          "high": "86",
          "low": "72",
        }
      }
      key="forecast30 Apr 2018"
    />
  </ul>
</div>
`;
